src/common/modals/InstanceManager/index.js

```diff
@@ -83,12 +145,53 @@ const menuEntries = {
   // settings: { name: "Settings", component: Overview },
   // servers: { name: "Servers", component: Overview }
 };
+
 const InstanceManager = ({ instanceName }) => {
   const dispatch = useDispatch();
+  const instancesPath = useSelector(_getInstancesPath);
   const [page, setPage] = useState(Object.keys(menuEntries)[0]);
   const instance = useSelector(state => _getInstance(state)(instanceName));
+  const [background, setBackground] = useState(instance?.background);
   const ContentComponent = menuEntries[page].component;
 
+  const updateBackGround = v => {
+    dispatch(
+      updateInstanceConfig(instanceName, prev => ({
+        ...prev,
+        background: v
+      }))
+    );
+  };
+
+  const openFileDialog = async () => {
+    const dialog = await ipcRenderer.invoke('openFileDialog', [
+      { name: 'Image', extensions: ['png', 'jpg', 'jpeg'] }
+    ]);
+    if (dialog.canceled) return;
+    const instancePath = path.join(instancesPath, instanceName);
+    const ext = path.extname(dialog.filePaths[0]);
+    const filePath = path.join(instancePath, `icon${ext}`);
+    await fss.copy(dialog.filePaths[0], filePath);
+
+    fs.readFile(filePath)
+      .then(res =>
+        setBackground(`data:image/png;base64,${res.toString('base64')}`)
+      )
+      .catch(console.warning);
+    setBackground(filePath);
+    updateBackGround(`icon${ext}`);
+  };
+
+  useEffect(() => {
+    if (instance?.background) {
+      fs.readFile(path.join(instancesPath, instanceName, instance?.background))
```

Comment:
There is no need to use ? here since you already check it in the upper condition